src/services/puzzle/puzzle-lib.ts

```diff
@@ -0,0 +1,21 @@
+import Config from "../../config.js";
+import Models from "../../database/models.js";
+import { PuzzleItem } from "database/puzzle-db.js";
+import { UpdateQuery } from "mongoose";
+
+/**
+ * Change's user puzzle.
+ * @param userId ID of the user to modify
+ * @param qid QuestionId (0-indexed) that was correctly submitted
+ * @returns Promise containing the puzzle
+ */
+export async function updatePuzzle(userId: string, qid: number): Promise<PuzzleItem | null> {
+    const updateQuery: UpdateQuery<PuzzleItem> = {
+        $set: {
+            lastCorrect: Math.floor(Date.now() / Config.MILLISECONDS_PER_SECOND),
+            [`problemComplete.${qid}`]: true,
```

Comment:
if you're using booleans as representations of T/F vals, then why do we need the variables `TRUE_VALUE` and `FALSE_VALUE`?

Reply:
Mongo takes 0, 1 and those count as magic numbers for the linter so it fails